src/common/Modal/index.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { FC, memo, useCallback, useEffect } from 'react'
+import { createPortal } from 'react-dom'
+import styled from 'styled-components'
+
+type PortalModalPropsType = {
+  visible: boolean
+  setIsModalVisible: (value: boolean) => void
+  children: React.ReactNode
+}
+
+export const Modal: FC<PortalModalPropsType> = memo(({ visible, children, setIsModalVisible }) => {
+  const onClickHandler = useCallback(() => {
+    setIsModalVisible(false)
+  }, [setIsModalVisible])
+  const root = document.querySelector('body')
+  useEffect(() => {
+    const onEscPressHandler = function (e: KeyboardEvent) {
```

Comment:
move this event handler outside of useEffect

Reply:
because there is no reason to keep it here. I mean this function will be recreated each time `setIsModalVisible` variable is changed